File: vol2/ch21/LeetCode343.cpp
class Solution {
public:
    int integerBreak(int n) {
        vector<int> dp(n+1,0);
        for(int i=1;i<=n;i++) {
            for(int j=1;j<i;j++) {
                dp[i]=max(dp[i],max((i-j)*dp[j],(i-j)*j));
            }
        }
        return dp[n];
    }
};

File: vol2/ch21/LeetCode343.py
class Solution:
    def integerBreak(self, n: int) -> int:
        dp=[0]*(n+1)
        for i in range(1,n+1):
            for j in range(1,i):
                dp[i]=max(dp[i],max((i-j)*dp[j],(i-j)*j))
        return dp[n]
